builder/amazon/common/step_run_spot_instance.go

```diff
@@ -260,17 +260,30 @@ func (s *StepRunSpotInstance) Run(ctx context.Context, state multistep.StateBag)
 				Tags:         spotTags,
 			},
 		}
+		launchTemplate.LaunchTemplateData.TagSpecifications = []*ec2.LaunchTemplateTagSpecificationRequest{
+			{
+				ResourceType: aws.String("instance"),
+				Tags:         spotTags,
+			},
+			{
+				ResourceType: aws.String("volume"),
+				Tags:         spotTags,
```

Comment:
I think this ought to be set from run_volume_tags. spot_tags is specific to the spot request itself rather than the resources created by the request.

Reply:
Yep, makes sense.